src/Microsoft.SqlTools.ServiceLayer/QueryExecution/ResultSet.cs

```diff
@@ -301,10 +381,10 @@ protected virtual void Dispose(bool disposing)
 
             IsBeingDisposed = true;
             // Check if saveTasks are running for this ResultSet
-            if (!saveTasks.IsEmpty)
+            if (!SaveTasks.IsEmpty)
             {
                 // Wait for tasks to finish before disposing ResultSet
-                Task.WhenAll(saveTasks.Values.ToArray()).ContinueWith((antecedent) =>
+                Task.WhenAll(SaveTasks.Values.ToArray()).ContinueWith((antecedent) =>
```

Comment:
Yes, I believe this was the reviewed design. We can change this to something else in the future or come up with something different for shutdown of the service versus disposal of the object.